### DISCUSSION TITLE
Set enviroment variable for solution via Github action

### DISCUSSION THREAD

**Discussion by LeonardSchwenk** [microsoft/powerplatform-actions#494] [Q&A] [answered]
Hello,
I want to insert different values into my solution specific environment variables.

Example:
I need a different value for the variable email_reciever in tst environment than in my prd environment for my solution .

How can I achieve this ?

Is there something like an input parameter for the github action ? I cannot find anything.

Or if it's not possible via the GitHub action, can I somehow achive this via find and replace when the solution is unpacked ?

![image](https://github.com/microsoft/powerplatform-actions/assets/26769569/f7360860-5624-4774-935e-e8704acda038)

**Comment by rpothin** [comment, marked as answer]
Hi @LeonardSchwenk,

I hope you are doing well.
I think what you are looking for is described in the following Microsoft documentation page: [Pre-populate connection references and environment variables for automated deployments](https://learn.microsoft.com/en-us/power-platform/alm/conn-ref-env-variables-build-tools)

Inside the [import-solution](https://github.com/microsoft/powerplatform-actions/blob/baa268c8cb72de054327df8fa55290ac4ace2bad/import-solution/action.yml#L78C1-L85C20) GitHub action, you can then set the "deployment-settings-file" input with the path of the relevant deployment settings file (*the one for the targeted environment*) in your repository. And don't forget to set the "use-deployment-settings-file" input to "true" 😉

I hope this will help you.
Have a great day.

Raph
